rep-0008.rst

```diff
@@ -22,27 +25,148 @@ Please follow Python's PEP 8 [1]_ for guidelines on formatting, naming, etc...
 The Loader
 ==========
 
-roslib is the only ROS Python package you can automatically assume to
+With ``catkin``
+---------------
+
+The loader method of accessing Python code in ROS is obsolete if you are using
+``catkin``.
+When using ``catkin``, the Python code for all packages will be on the
+``PYTHONPATH`` after sourcing the `setup.bash` file for your workspace.
+
+With ``rosbuild``
+-----------------
+
+``roslib`` is the only ROS Python package you can automatically assume to
 be importable other than your local files. If your Python package has
 dependencies, you must include the following header at the top::
 
     import roslib
     roslib.load_manifest('your_package_name')
 
-roslib.load_manifest looks at your manifest.xml and places your
-dependencies on your Python path. DO NOT use multiple load_manifest
+``roslib.load_manifest`` looks at your ``manifest.xml`` and places your
+dependencies on your Python path. DO NOT use multiple ``load_manifest``
 calls. If you need multiple calls, it's probably because you're
 missing the correct dependencies in your manifest.
 
 
-Package/Module Names (__init__.py files)
-========================================
+Package/Module Names (``__init__.py`` files)
+============================================
+
+With ``catkin``
+---------------
+
+When using ``catkin`` you can organize your Python code in any way that you like, but you must inform Python's ``setuptools`` where your code is located by creating a ``setup.py`` file.
+The recommended layout is to put all of your package's importable Python code in the ``src`` folder of your package and in a "Python package" with the same name as your package.
```

Comment:
Should we only recommend this layout or also mention the case with a `my_package` folder directly within the package folder?